src/solver/simulation/sim_alloc_probleme_hebdo.cpp

```diff
@@ -335,8 +334,10 @@ void SIM_AllocationConstraints(PROBLEME_HEBDO& problem,
             nbTimeSteps = 0;
             break;
         }
+        RESULTATS_CONTRAINTES_COUPLANTES res;
+        res.variablesDuales.assign(nbTimeSteps, 0.);
         if (nbTimeSteps > 0)
-            problem.ResultatsContraintesCouplantes[k].variablesDuales.assign(nbTimeSteps, 0.);
+            problem.ResultatsContraintesCouplantes.emplace(bc, res);
```

Comment:
You could change `RESULTATS_CONTRAINTES_COUPLANTES`'s constructor a bit to write this more compactly
```suggestion
if (nbTimeSteps > 0)
    problem.ResultatsContraintesCouplantes.emplace(bc, nbTimeSteps);
```

Full example
```cpp
#include <iostream>
#include <vector>
#include <map>

struct RESULT {
  RESULT(size_t size) : v_(size) {}
  std::size_t size() const {return v_.size();}
private:
  std::vector<int> v_;
};

int main() {
  std::map<int, RESULT> m;
  m.emplace(3, 4);
  for (auto& [k, v] : m)
    std::cout << k << " " << v.size() << std::endl;
  // 3 4
}
```